dnf-behave-tests/features/repoquery/main.feature

```diff
@@ -0,0 +1,521 @@
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom-a1-1:1.0-1.noarch
+      bottom-a1-1:1.0-1.src
+      bottom-a1-1:2.0-1.noarch
+      bottom-a1-1:2.0-1.src
+      top-a-2:2.0-2.src
+      top-a-2:2.0-2.x86_64
+      """
+
+Scenario: repoquery NAME-VERSION NAME-EPOCH:VERSION_GLOB-RELEASE
+ When I execute dnf with args "repoquery bottom-a1-1.0 top-a-1:[12].0-1"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom-a1-1:1.0-1.noarch
+      bottom-a1-1:1.0-1.src
+      top-a-1:1.0-1.src
+      top-a-1:1.0-1.x86_64
+      top-a-1:2.0-1.src
+      top-a-1:2.0-1.x86_64
+      """
+
+@xfail
+@bz1735687
+Scenario: repoquery NAME-VERSION NAME-EPOCH:VERSION_GLOB2-RELEASE
+ When I execute dnf with args "repoquery bottom-a1-1.0 top-a-1:[1-2].0-1"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom-a1-1:1.0-1.noarch
+      bottom-a1-1:1.0-1.src
+      top-a-1:1.0-1.src
+      top-a-1:1.0-1.x86_64
+      top-a-1:2.0-1.src
+      top-a-1:2.0-1.x86_64
+      """
+
+
+# --all: compatibility option, basically does nothing
+Scenario: dnf repoquery --all
+ When I execute dnf with args "repoquery"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom-a1-1:1.0-1.noarch
+      bottom-a1-1:1.0-1.src
+      bottom-a1-1:2.0-1.noarch
+      bottom-a1-1:2.0-1.src
+      bottom-a2-1:1.0-1.src
+      bottom-a2-1:1.0-1.x86_64
+      bottom-a3-1:1.0-1.src
+      bottom-a3-1:1.0-1.x86_64
+      bottom-a3-1:2.0-1.src
+      bottom-a3-1:2.0-1.x86_64
+      broken-deps-1:1.0-1.src
+      broken-deps-1:1.0-1.x86_64
+      mid-a1-1:1.0-1.src
+      mid-a1-1:1.0-1.x86_64
+      mid-a2-1:1.0-1.src
+      mid-a2-1:1.0-1.x86_64
+      top-a-1:1.0-1.src
+      top-a-1:1.0-1.x86_64
+      top-a-1:2.0-1.src
+      top-a-1:2.0-1.x86_64
+      top-a-2:2.0-2.src
+      top-a-2:2.0-2.x86_64
+      """
+
+Scenario: dnf repoquery --all NAME
+ When I execute dnf with args "repoquery --all top-a"
+ Then the exit code is 0
+  And stdout is
+      """
+      top-a-1:1.0-1.src
+      top-a-1:1.0-1.x86_64
+      top-a-1:2.0-1.src
+      top-a-1:2.0-1.x86_64
+      top-a-2:2.0-2.src
+      top-a-2:2.0-2.x86_64
+      """
+
+
+# --available is the default, scenarios above should cover it
+Scenario: dnf repoquery --available NAME
+ When I execute dnf with args "repoquery --available top-a-2.0"
+ Then the exit code is 0
+  And stdout is
+      """
+      top-a-1:2.0-1.src
+      top-a-1:2.0-1.x86_64
+      top-a-2:2.0-2.src
+      top-a-2:2.0-2.x86_64
+      """
+
+
+# --arch
+Scenario: repoquery --arch ARCH
+ When I execute dnf with args "repoquery --arch noarch"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom-a1-1:1.0-1.noarch
+      bottom-a1-1:2.0-1.noarch
+      """
+
+Scenario: repoquery --arch ARCH (nonexisting arch)
+ When I execute dnf with args "repoquery --arch yesarch"
+ Then the exit code is 0
+  And stdout is empty
+
+
+# --cacheonly
+Scenario: repoquery -C (without any cache)
```

Comment:
What about to add command `dnf clean all`?

Reply:
The cache is clean at the start of the scenario. I could add it to "make sure", but it is of no effect at this state and if someone ever changes the tests so that there is a cache at the start, some of the tests will actually start failing. So I don't think such change could go unnoticed and invalidate the tests.

Therefore it is perhaps even better to have tests that rely on the cache being clean at the start, since if anyone messed with this globally, there is a possibility they could be invalidating random other tests, which is not a good thing...